C++ compilation error with `exceptions:goto` + ARC/ORC when compiling basic branching code

### Description

This code fails to compile on devel and on 1.6.12 with `nim cpp --exceptions:goto --mm:orc file.nim`:


```nim
proc mysubstr: string =
  discard

proc findNormalized(x: string): int =
  discard

proc addfor =
  var x = findNormalized mysubstr()
  if x != 0: 
    echo "hi there :)"

addfor()
```

This is minimized from the original code sent to me in private from the Telegram (see https://t.me/nim_lang/139761 for the context). Minimization chain: `asynchttpserver` -> (dump async code) -> inline `asyncfutures` -> stacktrace generation code -> `strutils.%` -> `strutils.addf`

A "shallowly" minimized version is also available for making sure my repro didn't miss the original issue:
```nim
# example.nim
import asyncdispatch, asynchttpserver

let server = newAsyncHttpServer()

proc cb(req: Request): Future[void] {.async.} =
    discard

waitFor server.serve(Port(2060), cb, "127.0.0.1")
```

Both examples work with `--mm:refc` instead of ARC/ORC

### Nim Version

Nim Compiler Version 1.9.1 [Linux: amd64]
Compiled at 2023-03-28
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: 2315b01ae691e5e9e54fdfdfb4642c8fbc559e48
active boot switches: -d:release

### Current Output
With clang (for a better error)

```text

/home/dian/Projects/NimExperiments/here22/@mttt.nim.cpp:71:30: error: cannot jump from this goto statement to its label
        if (NIM_UNLIKELY(*nimErr_)) goto LA1_;
                                    ^
/home/dian/Projects/NimExperiments/here22/@mttt.nim.cpp:73:5: note: jump bypasses variable initialization
        NI x = T2_;
           ^
/home/dian/Projects/NimExperiments/here22/@mttt.nim.cpp:72:5: note: jump bypasses variable initialization
        NI T2_ = findNormalized__ttt_3(colontmpD_);
           ^
1 error generated.
Error: execution of an external compiler program 'clang++ -c -std=gnu++17 -funsigned-char  -pthread  -O3 -fno-ident   -I/home/dian/Things/nim/lib -I/home/dian/Projects/NimExperiments -o /home/dian/Projects/NimExperiments/here22/@mttt.nim.cpp.o /home/dian/Projects/NimExperiments/here22/@mttt.nim.cpp' failed with exit code: 1
```


### Expected Output

```text
Successful compilation
```


### Possible Solution

_No response_

### Additional Information

I also found out that https://github.com/nim-lang/Nim/issues/13998 fails to compile with `nim cpp --mm:orc --exceptions:goto file.nim` with the same error.

--- a/compiler/ccgstmts.nim
+++ b/compiler/ccgstmts.nim
@@ -329,7 +329,8 @@ proc genSingleVar(p: BProc, v: PSym; vn, value: PNode) =
   else:
     let imm = isAssignedImmediately(p.config, value)
     if imm and p.module.compileToCpp and p.splitDecls == 0 and
-        not containsHiddenPointer(v.typ):
+        not containsHiddenPointer(v.typ) and
+        nimErrorFlagAccessed notin p.flags:
       # C++ really doesn't like things like 'Foo f; f = x' as that invokes a
       # parameterless constructor followed by an assignment operator. So we
       # generate better code here: 'Foo f = x;'

